### PR TITLE
todo app from react1 week2 homework

### PR DESCRIPTION
**Pull request by aladdin91**
this is my homework for react1 week2

### CODE REVIEW

File: react/react1/week2/todo-app/src/components/TodoList.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  const mapTodo = () => {
+    if (todos.length !== 0) {
+      return todos.map((item) => (
+        <TodoItem key={item.id} item={item} todos={todos} setTodo={setTodo} />
+      ));
+    } else {
+      return <h3>No items</h3>;
+    }
+  };
```
Comment (shpomp) [inline react/react1/week2/todo-app/src/components/TodoList.jsx:16]: Nice approach to move this into a function. I would suggest to change the naming to a more typical "renderX" though. 
It makes sense that you are mapping the items and that is what you call it, but to somebody else working with the code it is not very informative from the first glance - mapping could be adding or recalculating properties, or something else. 
``renderTodos`` would clearly indicate what this is 👍🏽

File: react/react1/week2/todo-app/src/components/TodoList.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { useState } from "react";
+import TodoItem from "./TodoItems";
+import AddRandomTodo from "./AddRandomTodo";
+
+function ToDosList({ todoItem }) {
+  const [todos, setTodo] = useState(todoItem);
```
Comment (shpomp) [inline react/react1/week2/todo-app/src/components/TodoList.jsx:6]:
```suggestion
  const [todos, setTodos] = useState(todoItem);
```

Consistency 💡

File: react/react1/week2/todo-app/src/components/TodoItems.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+
+function TodoItem({ item, todos, setTodo }) {
+  const [checked, setChecked] = useState(false);
```
Comment (shpomp) [inline react/react1/week2/todo-app/src/components/TodoItems.jsx:4]: Would be much better to have this as the property of the todo item itself. Having this as a local state, it sets back to the default on every re-launch of the app. 
Try to add the ``completed`` property to the todo item and manage it that way 💡 👌🏽

**Review by shpomp — APPROVED** [review]
Good job! ⭐  👍🏽 
One suggestion about reformulation of the "completed" status functionality and some naming ideas. 
Otherwise very nicely done, good and cleanly implemented React basics 👏🏽

**Comment by aladdin91** [comment]
thanks a lot @shpomp for the review that was really helpful 🙏

**Pull request merged**